rfcs/0030-fork-signalling.md

```diff
@@ -0,0 +1,94 @@
+## Summary
+[summary]: #summary
+
+Nodes can signal to other nodes that they intend to switch from the current fork to a new fork.
+
+## Motivation
+[motivation]: #motivation
+
+Each node in the network has a notion of the current fork it's running on. A node may wish to
+change forks, perhaps because it's running new software, or to create a new subnet of nodes with
+some common interest. Such an ability is a limited form of on-chain governance. When a node
+proposes a new fork, other nodes need to be informed of the proposal.
+
+## Detailed design
+[detailed-design]: #detailed-design
+
+A fork is denoted by a fork identifier (`fork ID`). Blocks have two fork ID fields, one to
+indicate the current fork, another to indicate a proposed fork. Because there will usually
+not be a proposed fork, that field has an option type.
+
+When a node is started, the current fork ID is retrieved from the node's configuration.
+If the node is being run for the first time, the fork ID must be provided as a command-line
+flag. Alternatively, a current fork ID can be provided via a compile-time constant,
```

Comment:
I think this should be the only way. We shouldn’t require a command line flag for the first time.

Reply:
@bkase If so, is there a mechanism to change it from the compile-time value?

Reply:
Yes, I think compile time default with the command-line override seems right (to support the _really_ soft forks)